src/core/symbology/qgsarrowsymbollayer.cpp

```diff
@@ -215,7 +215,9 @@ void QgsArrowSymbolLayer::startRender( QgsSymbolRenderContext &context )
   mComputedHeadType = headType();
   mComputedArrowType = arrowType();
 
-  mSymbol->startRender( context.renderContext() );
+  mSymbol->setRenderHints( mSymbol->renderHints() | Qgis::SymbolRenderHint::IsSymbolLayerSubSymbol );
```

Comment:
Wouldn't be better to do it when adding the symbol to its parent symbol layer so the flag could be reused outside of the rendering pipeline?

Reply:
Let's defer that till it's actually needed. Turns out it's not so trivial to do, as the render hints aren't cloned (and I don't think they should be, either). The logic ends up much more complex as a result.